Sources/TuistGenerator/Mappers/LibraryResourcesProjectMapper.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// A project mapper that adds support for defining resources in  libraries
+public class LibraryResourcesProjectMapper: ProjectMapping {
+    public init() {}
+
+    public func map(project: Project) throws -> (Project, [SideEffectDescriptor]) {
+        var sideEffects: [SideEffectDescriptor] = []
+        var targets: [Target] = []
+
+        project.targets.forEach { target in
+            let (mappedTargets, targetSideEffects) = mapTarget(target, project: project)
+            targets.append(contentsOf: mappedTargets)
+            sideEffects.append(contentsOf: targetSideEffects)
+        }
+
+        return (project.with(targets: targets), sideEffects)
+    }
+
+    public func mapTarget(_ target: Target, project: Project) -> ([Target], [SideEffectDescriptor]) {
+        guard target.resources.count != 0 else { return ([target], []) }
+        var targets: [Target] = []
+
+        let bundleName = "\(target.name)Resources"
+        let (filePath, fileDescriptors) = synthesizedFile(bundleName: bundleName, target: target, project: project)
+        var modifiedTarget = target
+
+        if target.product.isStatic {
+            let resourcesTarget = Target(name: bundleName,
+                                         platform: target.platform,
+                                         product: .bundle,
+                                         productName: nil,
+                                         bundleId: "\(target.bundleId).resources",
+                                         resources: target.resources,
+                                         filesGroup: target.filesGroup)
+            modifiedTarget.resources = []
+            modifiedTarget.dependencies.append(.target(name: bundleName))
+            targets.append(resourcesTarget)
+        } else {}
+
+        modifiedTarget.sources.append((path: filePath, compilerFlags: nil))
```

Comment:
ouch, just realized this is adding a source file to any target that isn't static (e.g. adding to `.bundle` targets!)